client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,366 @@
+        pipeline._to_pipeline_configuration(pipeline_resources=step_resources)
+    assert (
+        str(err.value) == "pipeline_resources contains resources for a step not in the "
+        "pipeline: 'invalid_step_name'"
+    )
+    step_resources.pop("invalid_step_name")
+    # step name not a string
+    step_resources.update({123: resources})
+    with pytest.raises(TypeError) as err2:
+        pipeline._to_pipeline_configuration(pipeline_resources=step_resources)
+    assert (
+        str(err2.value) == "pipeline_resources keys must be type str, not <class 'int'>"
+    )
+    step_resources.pop(123)
+    # step resource not a Resources object
+    step_resources.update({"step_1": "not_resources"})
+    with pytest.raises(TypeError) as err3:
+        pipeline._to_pipeline_configuration(pipeline_resources=step_resources)
+    assert (
+        str(err3.value)
+        == "pipeline_resources values must be type Resources, not <class 'str'>"
+    )
+
+
+def test_to_pipeline_configuration_no_resources(
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that a pipeline configuration can be constructed from a
+    RegisteredPipeline object without providing any pipeline resources.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rm")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+    pipeline_configuration = pipeline._to_pipeline_configuration()
+    assert pipeline_configuration["pipeline_version_id"] == pipeline.id
+    for graph_step, config_step in zip(graph.steps, pipeline_configuration["steps"]):
+        # All steps are included in the configuration
+        assert graph_step.name == config_step["name"]
+        # No resources are found in the resulting configuration
+        assert "resources" not in config_step.keys()
+
+
+def test_from_pipeline_definition(
+    make_mock_registered_model_version,
+    mocked_responses,
+) -> None:
+    """Test that a RegisteredPipeline object can be constructed from a pipeline
+    definition.
+
+    The model version's `_get_artifact` function is overidden in the
+    mocked RMV fixture to return a simple, consistent pipeline definition.
+    Calls related to the fetching of the RMV and RM are mocked.
+    """
+    mocked_responses.get(
+        "https://test_socket/api/v1/registry/model_versions/1",
+        json={},
+        status=200,
+    )
+    mocked_responses.get(
+        "https://test_socket/api/v1/registry/model_versions/2",
+        json={},
+        status=200,
+    )
+    mocked_responses.get(
+        "https://test_socket/api/v1/registry/registered_models/0",
+        json={},
+        status=200,
+    )
```

Comment:
Let's fetch socket and scheme from `rmv`

Reply:
Fixed